perses/tests/test_relative.py

```diff
@@ -823,16 +823,18 @@ def flattenedHybridTopologyFactory_energies(topology, chain, system, positions,
         # Make list of off atoms that should have flattened torsions/exceptions
         off_atoms = topology_proposal.unique_new_atoms if endstate == 0 else topology_proposal.unique_old_atoms
         system = topology_proposal.old_system if endstate == 0 else topology_proposal.new_system
+        # TODO will openmm ever allow duplicate names here?
+        force_names = {force.__class__.__name__ : index for index, force in enumerate(system.getForces())}
```

Comment:
FYI: There's now a `force.getName()` method that can be used instead, though this is a recent addition.
The name can even be set by users if nonstandard names are desired.